### PR TITLE
test(simulation): re-enable simulation tests

### LINKED ISSUE #1745: test app import/export

### PR DESCRIPTION
**Pull request by k-yang**
# Purpose / Abstract

Re-enables and fixes the following simulation tests:

- TestAppImportExport
- TestAppSimulationAfterImport
- TestAppStateDeterminism
- TestFullAppSimulation

The tests were copied over from https://github.com/cosmos/cosmos-sdk/blob/d09f40d0a1160dc576c68b6be9c54e4ac07aa92f/simapp/sim_test.go and adapted to run with the Nibiru app.

The PR also adds a GHA workflow to run the simulation tests once a day on the `main` branch.

There are follow-up PRs to add simulation operations for Nibiru custom modules (see #1737)

- Closes #1745, #1746, #1747, #1748

## Summary by CodeRabbit

This update does not change the existing bullet-point list. Therefore, the previous list remains valid and can be repeated verbatim:

## Summary by CodeRabbit

- **New Features**
  - Introduced a workflow for automated simulation tests.
  - Enhanced simulation test commands and parameters for better testing scenarios.
  - Expanded application initialization and configuration options in testing utilities.

- **Improvements**
  - Renamed a key field for clarity and consistency across the application codebase.
  - Updated module management calls to align with the renamed field.

- **Documentation**
  - Updated CHANGELOG with recent pull request modifications and dependency updates.

- **Refactor**
  - Restructured simulation-related code to improve maintainability and performance.

- **Bug Fixes**
  - Addressed issues in simulation tests to ensure reliability.

### CODE REVIEW

File: x/sudo/module.go [hunk 1]
```diff
@@ -151,3 +154,34 @@ func (am AppModule) BeginBlock(_ sdk.Context, _ abci.RequestBeginBlock) {}
...
+//----------------------------------------------------------------------------
+
+// GenerateGenesisState implements module.AppModuleSimulation.
+func (AppModule) GenerateGenesisState(simState *module.SimulationState) {
+	simulation.RandomizedGenState(simState)
+}
+
+// RegisterStoreDecoder implements module.AppModuleSimulation.
+func (AppModule) RegisterStoreDecoder(sdk.StoreDecoderRegistry) {
+}
```
[the remark this answers is not in the document]
Comment (k-yang) [reply x/sudo/module.go:169]: I will implement these in a follow-up PR (see #1735)

[the remark this answers is not in the document]
Comment (k-yang) [reply x/sudo/module.go:169]: The issue is actually #1736

File: app/app.go [hunk 2]
```diff
@@ -90,7 +90,7 @@ type NibiruApp struct {
 	AppKeepers // embed all module keepers
 
 	// the module manager
-	mm *module.Manager
+	ModuleManager *module.Manager
```
Comment (jgimeno) [inline app/app.go:93]: do we have some benefit of adding it public?

**Review by Unique-Divine — APPROVED** [review]

File: app/app.go (hunk 2 above)
Comment (k-yang) [reply app/app.go:93]: It's required for `TestAppImportExport`.

**Pull request merged**